### ISSUE TITLE
Putout doesn't work with Vue

### ISSUE THREAD

**Issue by dotai2012** [coderaiser/putout#66]
ESlint version: 7.3.0
Bug: Putout doesn't work with Vuejs

![Capture](https://user-images.githubusercontent.com/12532557/124830325-b65aad00-df47-11eb-96fd-d0c226a864cd.PNG)

When I comment out "putout" in the plugins's array, it works fine

This is my eslint's config file

```json
{
  "env": {
    "browser": true,
    "es2020": true,
    "jest/globals": true,
    "jquery": true,
    "node": true,
    "react-native/react-native": true
  },
  "extends": [
    "airbnb",
    "airbnb-typescript",
    "airbnb/hooks",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:json/recommended",
    "plugin:jsx-a11y/recommended",
    "plugin:lodash/recommended",
    "plugin:node/recommended",
    "plugin:optimize-regex/recommended",
    "plugin:react-hooks/recommended",
    "plugin:react-native/all",
    "plugin:react-redux/recommended",
    "plugin:react/recommended",
    "plugin:vue/vue3-recommended",
    "plugin:vue-scoped-css/vue3-recommended",
    "plugin:security/recommended",
    "plugin:sonarjs/recommended",
    "plugin:unicorn/recommended",
    "plugin:import/errors",
    "plugin:import/typescript",
    "plugin:array-func/recommended",
    "plugin:@getify/proper-arrows/getify-says",
    "plugin:no-unsanitized/DOM",
    "plugin:putout/recommended"
  ],
  "overrides": [
    {
      "files": [
        "**/*.ts",
        "**/*.tsx"
      ],
      "parser": "@typescript-eslint/parser",
      "rules": {
        "@typescript-eslint/ban-ts-ignore": 0,
        "@typescript-eslint/explicit-function-return-type": "error",
        "@typescript-eslint/explicit-module-boundary-types": "error",
        "@typescript-eslint/interface-name-prefix": 0,
        "@typescript-eslint/no-explicit-any": 0,
        "@typescript-eslint/no-non-null-assertion": 0,
        "@typescript-eslint/no-var-requires": "error"
      }
    },
    {
      "files": [
        "**/*.vue"
      ],
      "parser": "vue-eslint-parser",
      "parserOptions": {
        "parser": "@typescript-eslint/parser"
      }
    }
  ],
  "parser": "@babel/eslint-parser",
  "parserOptions": {
    "ecmaVersion": "latest",
    "jsx": true,
    "requireConfigFile": false,
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint",
    "html",
    "json",
    "json-format",
    "jsx-a11y",
    "lodash",
    "node",
    "optimize-regex",
    "react",
    "react-hooks",
    "react-native",
    "react-redux",
    "security",
    "sonarjs",
    "unicorn",
    "import",
    "array-func",
    "@getify/proper-arrows",
    "jest",
    "putout"
  ],
  "rules": {
    "@getify/proper-arrows/name": 0,
    "@getify/proper-arrows/params": 0,
    "@getify/proper-arrows/return": 0,
    "@getify/proper-arrows/this": 0,
    "@getify/proper-arrows/where": 0,
    "@typescript-eslint/ban-ts-comment": 0,
    "@typescript-eslint/dot-notation": 0,
    "@typescript-eslint/explicit-function-return-type": 0,
    "@typescript-eslint/explicit-module-boundary-types": 0,
    "@typescript-eslint/naming-convention": 0,
    "@typescript-eslint/no-implied-eval": 0,
    "@typescript-eslint/no-throw-literal": 0,
    "@typescript-eslint/no-unused-expressions": 0,
    "@typescript-eslint/no-var-requires": 0,
    "@typescript-eslint/return-await": 0,
    "@typescript-eslint/space-before-function-paren": [
      "error"
    ],
    "@typescript-eslint/triple-slash-reference": 0,
    "camelcase": 0,
    "func-names": 0,
    "global-require": 0,
    "import/extensions": [
      "error",
      "ignorePackages",
      {
        "js": "never",
        "jsx": "never",
        "ts": "never",
        "tsx": "never"
      }
    ],
    "import/no-dynamic-require": 0,
    "import/no-extraneous-dependencies": 0,
    "import/no-named-as-default-member": 0,
    "import/prefer-default-export": 0,
    "indent": 0,
    "jest/no-disabled-tests": "warn",
    "jest/no-focused-tests": "error",
    "jest/no-identical-title": "error",
    "jest/prefer-to-have-length": "warn",
    "jest/valid-expect": "error",
    "jsx-a11y/click-events-have-key-events": 0,
    "jsx-a11y/label-has-associated-control": 0,
    "jsx-a11y/no-static-element-interactions": 0,
    "linebreak-style": 0,
    "lodash/chaining": 0,
    "lodash/import-scope": 0,
    "lodash/prefer-constant": 0,
    "lodash/prefer-immutable-method": 0,
    "lodash/prefer-lodash-method": 0,
    "lodash/prefer-some": 0,
    "max-classes-per-file": 0,
    "max-len": 0,
    "new-cap": 0,
    "no-async-promise-executor": 0,
    "no-console": 0,
    "no-extend-native": 0,
    "no-param-reassign": 0,
    "no-plusplus": 0,
    "no-process-exit": 0,
    "no-restricted-properties": 0,
    "no-underscore-dangle": 0,
    "node/no-missing-import": 0,
    "node/no-unpublished-import": 0,
    "node/no-unpublished-require": 0,
    "node/no-unsupported-features/es-syntax": 0,
    "putout/destructuring-as-function-argument": 0,
    "putout/keyword-spacing": 0,
    "putout/newline-function-call-arguments": 0,
    "putout/putout": [
      "error",
      {
        "plugins": [
          "remove-duplicate-interface-keys"
        ],
        "rules": {
          "apply-numeric-separators": "off",
          "convert-for-each-to-for-of": "off",
          "promises/add-missing-await": "off",
          "remove-console": "off",
          "remove-constant-conditions": "off",
          "remove-empty/import": "off",
          "remove-unused-types": "off",
          "remove-unused-variables": "off",
          "split-nested-destructuring": "off",
          "strict-mode/add": "off"
        }
      }
    ],
    "quote-props": 0,
    "react-native/no-color-literals": 0,
    "react-native/no-raw-text": 0,
    "react-native/no-single-element-style-arrays": 0,
    "react/destructuring-assignment": 0,
    "react/display-name": 0,
    "react/forbid-prop-types": 0,
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [
          ".js",
          ".jsx",
          ".tsx"
        ]
      }
    ],
    "react/jsx-fragments": 0,
    "react/jsx-props-no-spreading": 0,
    "react/no-access-state-in-setstate": 0,
    "react/prop-types": 0,
    "react/require-default-props": 0,
    "react/state-in-constructor": 0,
    "react/style-prop-object": 0,
    "security/detect-child-process": 0,
    "security/detect-non-literal-fs-filename": 0,
    "security/detect-non-literal-regexp": 0,
    "security/detect-object-injection": 0,
    "security/detect-possible-timing-attacks": 0,
    "sonarjs/cognitive-complexity": 0,
    "sonarjs/no-small-switch": 0,
    "space-before-function-paren": "off",
    "unicorn/catch-error-name": 0,
    "unicorn/filename-case": 0,
    "unicorn/no-array-for-each": 0,
    "unicorn/no-array-reduce": 0,
    "unicorn/no-process-exit": 0,
    "unicorn/no-reduce": 0,
    "unicorn/no-useless-undefined": 0,
    "unicorn/numeric-separators-style": 0,
    "unicorn/prefer-module": 0,
    "unicorn/prefer-node-protocol": 0,
    "unicorn/prefer-spread": 0,
    "unicorn/prevent-abbreviations": 0
  },
  "settings": {
    "import/resolver": {
      "node": {
        "extensions": [
          ".js",
          ".jsx",
          ".ts",
          ".tsx"
        ]
      }
    },
    "json/ignore-files": [
      "**/package-lock.json"
    ],
    "json/json-with-comments-files": [
      "**/tsconfig.json",
      ".vscode/**"
    ],
    "json/sort-package-json": "standard",
    "react": {
      "version": "latest"
    }
  }
}
```

**Comment by coderaiser** [comment]
That's right, `vue` isn't supported by `putout` (and there is no information about `vue` in documentation). I see that there is [vue-eslint-parser](https://github.com/vuejs/vue-eslint-parser) with outdated dependencies (like `espree`). Unfortunately I nee to [parse the source after eslint](https://github.com/coderaiser/putout/blob/v18.4.0/packages/eslint-plugin-putout/lib/putout/index.js#L50-L52) to receive [babel ast](https://github.com/babel/babel/blob/main/packages/babel-parser/ast/spec.md). I can't use [estree-to-babel](https://github.com/coderaiser/estree-to-babel) because `eslint tests` shows:

```
AssertionError [ERR_ASSERTION]: Rule should not modify AST.
```

But `Putout` modifies `AST`, this is how it works. [Structured Clone](https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript/10916838#10916838) can be used for this purpose.
I'm wandering if it can affect parsing of typescript files, as I see [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/typescript-estree#ast-alignment-tests) uses same `AST` nodes as `Babel`.

Could you please provide me code example, as I understand your project has both `vue` and `typescript`.

**Comment by coderaiser** [comment]
Landed in `eslint-plugin-putout@8.1.0` 🎉 . Is it works for you?

**Comment by dotai2012** [comment]
> Landed in `eslint-plugin-putout@8.1.0` 🎉 . Is it works for you?

Yes, it works. Thank you very much :D
